bundles/org.openhab.binding.smgw/src/main/java/org/openhab/binding/smgw/internal/SmgwHandlerFactory.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smgw.internal;
+
+import static org.openhab.binding.smgw.internal.SmgwBindingConstants.*;
+
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.scheduler.CronScheduler;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link SmgwHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ThingHandlerFactory.class)
+public class SmgwHandlerFactory extends BaseThingHandlerFactory {
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_SMGW);
+    private final HttpClient httpClient;
+    private final CronScheduler cronScheduler;
+
+    @Activate
+    public SmgwHandlerFactory(@Reference HttpClientFactory clientFactory, @Reference CronScheduler cronScheduler)
+            throws Exception {
+        httpClient = clientFactory.createHttpClient("smgw", new SslContextFactory.Client(true));
+        httpClient.start();
```

Comment:
The service will not start and an error logged. Since it'll not work without a started client, it doesn't make sense to activate a component. The HTTP binding catches the exception and throws a `RuntimeException`.

Reply:
Thank you for the explanation.
Probably something to replicate in every binding starting a dedicated HTTP client.

Reply:
Another benefit is that in the case the activate method (or the constructor) throws an exception, Karaf will try to re-add the service later.